src/Razor/src/Microsoft.VisualStudio.RazorExtension.Dependencies/AssemblyCodeBases.cs

```diff
@@ -80,12 +80,24 @@
     OldVersionLowerBound = "4.4.0.0",
     OldVersionUpperBound = "4.6.0.0",
     NewVersion = "4.6.0.0")]
+[assembly: ProvideBindingRedirection(
+    AssemblyName = "Microsoft.CodeAnalysis.VisualBasic",
+    GenerateCodeBase = true,
+    OldVersionLowerBound = "4.4.0.0",
+    OldVersionUpperBound = "4.6.0.0",
+    NewVersion = "4.6.0.0")]
```

Comment:
These being manually maintained feels wrong because there are several error paths here: 

1. As this change shows it's easy to forget individual libraries.
2. When the integration image updates to a new version we are broken until these values are updated (pretty sure). 

Did we ever consider a more automated way of maintaining these?

Reply:
Yes, this is very problematic, and both of your points are dead on.

1. This isn't even "forgetting", this change is because there happened to be an incompatible change in this specific Roslyn DLL, between whats in VS Preview and what we reference, so the list of DLLs here is not exhaustive, its just the ones that have caused problems so far.
2. Sam has tweaks to this in a PR which look like they will help, so there should be some improvements here I can follow up with, at least as far as version number maintenance goes.

I think the only full "solution" to this is, which would allow us to remove the dependencies project altogether is to utilize main-vs-deps, however with EA to Roslyn and potentially tweaks to CLaSP or the Roslyn LSP server, this would likely be permanent, and get used a reasonable amount for regular coding. We'd also need to be strict about only allowing main-vs-deps to reference a Roslyn version that is already inserted into a VS build that is available as an image, so even then our velocity would be severely hampered.

Reply:
https://github.com/dotnet/razor/pull/8410 simplifies this and addresses point 2. If the build is green, hopefully it buys us a reasonable amount of time.